flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -126,17 +125,4 @@ public void validateColumnListParams(
         // this makes it possible to ignore them in the validator and fall back to regular row types
         // see also SqlFunction#deriveType
     }
-
```

Comment:
Why have you removed this code?

Reply:
I see the issue. I wonder if the proper nested validation of explain and statement set can happen here as well, rather than on the planner side, as this seems the place where the validation belongs to. But I might be mistaken here.

Reply:
I didn't get the confusion " validation code scattered through 2/3 different components of the codebase", I just added the validation in the PlannerImpl. I add cases in some part of  planner tests, because we need to make sure that we really support the new syntax e2e, just like other kind of sql statements. 

I reverted the change you made in FLINK-22942, because I think the change was made in the wrong way, and it is necessary in the validation in StatementSet and Execute, that's why I made the change in this pr:
1.  we should not call validateInsert directly because some metadata used in validation would missed.
2. the former change didn't cover explain
3. in my understanding, we put the validation of flink custom node in the PlannerImpl, such as SqlRichExplain/RichSqlInsert, because calcite validator cannot cover them, and we wan to put all of validation in one place. Another way is put the validation in the SqlRichExplain/RichSqlInsert by rewriting the SqlNode#validate, but it would make the validation scattering in every node.

Reply:
I think it's all right to keep as you've done, and thanks for fixing FLINK-22942 as well.